application/models/FleetModel.php

```diff
@@ -45,12 +52,33 @@ public function all()
     public function get($id, $key2 = null)
     {
         $plane = parent::get($id, $key2);
-        $wackyPlane = json_decode(file_get_contents(WACKY_SERVER_URI_BASE . '/airplanes/' . $plane->airplaneCode));
+        $wackyPlane = $this->wackyModel->getAirplane($plane->airplaneCode);
+        $wackyPlane->id = $id;
+        return $wackyPlane;
+    }
 
-        foreach (get_object_vars($wackyPlane) as $prop => $val)
-            if (!property_exists($plane, $prop))
-                $plane->$prop = $val;
+    public function validFleet($plane) {
+        return $this->validBudget($plane);
+    }
 
-        return $plane;
+    /*
+    * Validator for the planes in the fleet.
+    * param $fleet - Array representing the airline's fleet
+    * A fleet will be considered invalid and returns false if:
+    *   - The sum of the cost the planes in the fleet is overbudget (10 million)
+    */
+    public function validBudget($plane) {
+        $fleet = $this->all();
+        $fleetPrice = 0;
+
+        foreach($fleet as $kitePlane)
+        {
+            $wackyPlane = $this->wackyModel->getAirplane($kitePlane[key]);
```

Comment:
I'm not sure we need another call to get the plane info from the wacky server. `$this->all()` should have given back the price already.

Reply:
fixed